homeassistant/components/binary_sensor/deconz.py

```diff
@@ -0,0 +1,91 @@
+"""
+Support for deCONZ binary sensor support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor/deconz/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.deconz import (
+    ATTR_FW_VERSION, ATTR_MANUFACTURER, ATTR_MODEL_ID, ATTR_UNIQUE_ID,
+    DECONZ_DATA, DOMAIN)
+from homeassistant.const import ATTR_BATTERY_LEVEL
+from homeassistant.core import callback
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup binary sensor platform for Deconz."""
+    from pydeconz.sensor import DECONZ_BINARY_SENSOR
+    sensors = hass.data[DECONZ_DATA].sensors
+
+    for _, sensor in sensors.items():
+        if sensor.type in DECONZ_BINARY_SENSOR:
+            async_add_devices([DeconzBinarySensor(sensor)], True)
+
+
+class DeconzBinarySensor(BinarySensorDevice):
+    """Representation of a binary sensor."""
+
+    def __init__(self, sensor):
+        """Setup sensor and add update callback to get data from websocket."""
+        self._sensor = sensor
+        self._sensor.register_callback(self._update_callback)
+
+    @callback
```

Comment:
This is not a callback in the Home Assistant async sense. This only works with `hass.async_add_job` / `hass.add_job`

Reply:
I double checked with @armills comment on my latest Axis PR and his comment was "The datagram endpoint will be called on the event loop, so it's safe to annotate it as @callback"

Reply:
Also regarding: https://github.com/home-assistant/home-assistant/pull/10321#discussion_r152920721

Looking through the upstream lib it does look like this should all be scheduled properly on the hass event loop. We should indicate it as such, although I personally don't have any preference between `@callback`, `async_`, or just a docstring note. I'd personally prefer keeping the decorator and adding the `async_` prefix. Even if the decorator is non-functional as it's currently used, it's nice to have the semantic information attached to the function if we ever used it differently.

Breadcrumbs:
https://github.com/Kane610/deconz/blob/1ca6c5fa6dbbe1a796282a13b602c6a85735fb33/pydeconz/websocket.py#L97
https://github.com/Kane610/deconz/blob/1ca6c5fa6dbbe1a796282a13b602c6a85735fb33/pydeconz/__init__.py#L37
https://github.com/Kane610/deconz/blob/1ca6c5fa6dbbe1a796282a13b602c6a85735fb33/pydeconz/__init__.py#L130
https://github.com/Kane610/deconz/blob/1ca6c5fa6dbbe1a796282a13b602c6a85735fb33/pydeconz/deconzdevice.py#L46